testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java

```diff
@@ -1250,6 +1257,20 @@ public void testDataMappingSmokeTest(DataMappingTestSetup dataMappingTestSetup)
         return Optional.of(dataMappingTestSetup);
     }
 
+    protected boolean isTestingCaseSensitiveData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getTrinoTypeName();
+        if (!typeName.contains("char")) {
+            return false;
+        }
+        return isUpperCase(dataMappingTestSetup.getSampleValueLiteral()) ^ isUpperCase(dataMappingTestSetup.getHighValueLiteral());
```

Comment:
Please avoid using `^` on booleans. `!=` might be a replacement.

However, it would be better to enumerate the cases here explicitly, or narrow the condition here.
-- We don't this method to return true when test data contains eg `Egypt` and `Alabama`.